### PR TITLE
test: Tests and Storybook entry for the IndeterminateCheckbox Component

### PR DESCRIPTION
**Pull request by geido**
Adds the test cases and a storybook entry with controls for the IndeterminateCheckbox component, while moving it to its dedicated directory.

### CODE REVIEW

File: superset-frontend/src/components/IndeterminateCheckbox/IndeterminateCheckbox.test.tsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+const mockedProps = {
+  checked: false,
+  id: 'checkbox-id',
+  indeterminate: false,
+  title: 'Checkbox title',
+  onChange: () => null,
+};
+
+jest.mock('../Icon', () => ({
```
Comment (michael-s-molina) [inline superset-frontend/src/components/IndeterminateCheckbox/IndeterminateCheckbox.test.tsx:33]: @geido I think you shouldn't mock the `Icon`. It's an integral part of this component, and you are making assertions about it. For example:

```
test('should render the checkbox-half icon', () => {
  const indeterminateProps = {
    ...mockedProps,
    indeterminate: true,
  };
  render(<IndeterminateCheckbox {...indeterminateProps} />);
  expect(screen.getByTestId('icon')).toBeInTheDocument();
  expect(screen.getByTestId('icon')).toHaveAttribute(
    'data-name',
    'checkbox-half',
  );
});
```
The attribute `data-name` only exists in your mock, but not in the real component. 

Take a look at this [PR](https://github.com/apache/superset/pull/13457). Here I added `role` and `aria-label` properties to the `Icon` component. You could base all your queries using these props. Unfortunately, it's not merged yet, but you can copy and paste it here.

If you prefer to mock the `Icon`, I would suggest only making assertions about properties that also exist on the real component 😉

**Review by michael-s-molina — APPROVED** [review]
LGTM

File: superset-frontend/src/components/IndeterminateCheckbox/IndeterminateCheckbox.test.tsx [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+import { supersetTheme } from '@superset-ui/core';
+import userEvent from '@testing-library/user-event';
+import IndeterminateCheckbox from '.';
+
+const mockedProps = {
+  checked: false,
+  id: 'checkbox-id',
+  indeterminate: false,
+  title: 'Checkbox title',
+  onChange: () => null,
```
Comment (yardz) [inline superset-frontend/src/components/IndeterminateCheckbox/IndeterminateCheckbox.test.tsx:30]: I think it would be nice to put a `jest.fn()` here

**Review by yardz — APPROVED** [review]

**Review by pkdotson — APPROVED** [review]
LGTG!

**Pull request merged**